tests/loggers/test_mosaicml_logger.py

```diff
@@ -194,3 +196,88 @@ def test_auto_add_logger(monkeypatch, platform_env_var, access_token_env_var, lo
+                                                    sequence_length=8,
+                                                    use_keys=True)
+    collator = transformers.DataCollatorForLanguageModeling(tokenizer=tiny_bert_tokenizer, mlm_probability=0.15)
+    dataloader = DataLoader(pretraining_train_dataset,
+                            batch_size=4,
+                            sampler=dist.get_sampler(pretraining_train_dataset),
+                            collate_fn=collator)
+    trainer = Trainer(model=SimpleTransformerMaskedLM(vocab_size=tiny_bert_tokenizer.vocab_size),
+                      train_dataloader=dataloader,
+                      max_duration='64tok',
+                      loggers=[MosaicMLLogger()])
+    trainer.fit()
+    assert 'mosaicml/training_progress_unit' in mock_mapi.run_metadata[run_name]
+    assert mock_mapi.run_metadata[run_name]['mosaicml/training_progress_unit'] == str(TimeUnit.TOKEN)
+    assert 'mosaicml/training_progress' in mock_mapi.run_metadata[run_name]
+    assert mock_mapi.run_metadata[run_name]['mosaicml/training_progress'] == '[token=64/64]'
+    assert 'mosaicml/training_sub_progress' not in mock_mapi.run_metadata[run_name]
+
+
+def test_batch_training_progress_logged(monkeypatch):
+    mock_mapi = MockMAPI()
+    monkeypatch.setattr(mcli, 'update_run_metadata', mock_mapi.update_run_metadata)
+    run_name = 'test-run-name'
+    monkeypatch.setenv('RUN_NAME', run_name)
+    trainer = Trainer(model=SimpleModel(),
+                      train_dataloader=DataLoader(RandomClassificationDataset()),
+                      train_subset_num_batches=1,
+                      max_duration='4ba',
+                      loggers=[MosaicMLLogger()])
+    trainer.fit()
+    assert 'mosaicml/training_progress_unit' in mock_mapi.run_metadata[run_name]
+    assert mock_mapi.run_metadata[run_name]['mosaicml/training_progress_unit'] == str(TimeUnit.BATCH)
+    assert 'mosaicml/training_progress' in mock_mapi.run_metadata[run_name]
+    assert mock_mapi.run_metadata[run_name]['mosaicml/training_progress'] == '[batch=4/4]'
+    assert 'mosaicml/training_sub_progress' not in mock_mapi.run_metadata[run_name]
+
+
+def test_epoch_training_progress_logged(monkeypatch):
+    mock_mapi = MockMAPI()
+    monkeypatch.setattr(mcli, 'update_run_metadata', mock_mapi.update_run_metadata)
+    run_name = 'test-run-name'
+    monkeypatch.setenv('RUN_NAME', run_name)
+    # Because batch timestamp resets to 0 at the end of the train, we use in memory logger to check an intermediate epoch timestamp
+    in_memory_logger = InMemoryLogger()
+    trainer = Trainer(model=SimpleModel(),
+                      train_dataloader=DataLoader(RandomClassificationDataset(), batch_size=2),
+                      train_subset_num_batches=2,
+                      max_duration='2ep',
+                      loggers=[MosaicMLLogger(), in_memory_logger])
+    trainer.fit()
+    assert 'mosaicml/training_progress_unit' in mock_mapi.run_metadata[run_name]
```

Comment:
nit: use a variable for mock_mapi.run_metadata[run_name] because it's referenced multiple times